examples/example-hono-api/src/auth0.ts

```diff
@@ -0,0 +1,166 @@
+import type {Context} from 'hono';
+import {env} from 'hono/adapter';
+import type {MiddlewareHandler} from 'hono/types';
+import {ApiClient} from '@auth0/auth0-api-js';
+import {HTTPException} from 'hono/http-exception';
+import {JWTPayload} from 'jose';
+
+export type JwtEnv = {
+    Variables: {
+        user: JWTPayload
+    }
+};
+
+export const jwt = (
+    options?: {
+        auth0_domain: string;
+        auth0_audience: string;
+    },
+): MiddlewareHandler => {
+
+    let apiClient: ApiClient | null = null;
+
+    return async function jwt(ctx, next) {
+
+        // 1. Create an instance of ApiClient and pass it the configuration
+        if (!apiClient) {
+            const auth0Env = env(ctx);
+            const {auth0_domain, auth0_audience} = options || {
+                auth0_domain: auth0Env.AUTH0_DOMAIN,
+                auth0_audience: auth0Env.AUTH0_AUDIENCE,
+            };
+            if (!auth0_domain || auth0_domain.length === 0) {
+                throw new Error('Auth0 middleware requires options "auth0_domain"');
+            }
+            if (!auth0_audience || auth0_audience.length === 0) {
+                throw new Error('Auth0 middleware requires options "auth0_audience"');
+            }
+
+            apiClient = new ApiClient({
+                domain: auth0_domain,
+                audience: auth0_audience,
+            });
```

Comment:
Any reason not to stick to avoiding snake casing here and go with the below?

```suggestion
            const { domain, audience } = options || {
                domain: auth0Env.AUTH0_DOMAIN,
                audience: auth0Env.AUTH0_AUDIENCE,
            };
            if (!domain || domain.length === 0) {
                throw new Error('Auth0 middleware requires options "domain"');
            }
            if (!audience || audience.length === 0) {
                throw new Error('Auth0 middleware requires options "audience"');
            }

            apiClient = new ApiClient({
                domain,
                audience,
            });
```

If we do this, we should also change line 16 to 17.